### ISSUE TITLE
small objects detection 

### ISSUE THREAD

**Issue by merbenf** [sovit-123/fasterrcnn-pytorch-training-pipeline#140]
hello again it finally all works thank you for this valuable repo but i just wanted to ask for "sahi" model wich is used for small objects detection how can i use it in inference with faster r cnn and if there any other models are used for very small (tiny) objects to be used . 
thank you all .

**Comment by sovit-123** [comment]
Hello @merbenf 
I have not integrated SAHI into the project yet. I will add that to my TODO list.

**Comment by merbenf** [comment]
> Hello @merbenf I have not integrated SAHI into the project yet. I will add that to my TODO list.

thank you sir we are waiting for it

**Comment by joshua-atolagbe** [comment]
@sovit-123 , any update about this?

**Comment by sovit-123** [comment]
I will be working on this next.

**Comment by joshua-atolagbe** [comment]
Okay great. Looking forward to this

**Comment by sovit-123** [comment]
@joshua-atolagbe 
I have SAHI image inference for all the pretrained torchvision models with almost all the arguments similar to `inference.py`.
The code is in the `sahi_inference.py` file.
You can find the [SAHI branch here](https://github.com/sovit-123/fasterrcnn-pytorch-training-pipeline/tree/sahi).

I will merge it after a bit more testing.

**Comment by joshua-atolagbe** [comment]
You’re the best bro. Will check it out script. I’m hoping it will also calculate the metrics like map50, precision, recall and fps too

**Comment by sovit-123** [comment]
@joshua-atolagbe 
Thanks.
Just for some clarification. It is the inference script. It will give you the FPS on the folder of images (the directory path that you will pass that contains the images). However, it does not carry out the metric calculations. It is quite difficult to integrate SAHI with object detection evaluation. Although it sounds plausible to add that, it is not very straightforward.

However, the image inference with SAHI is in working state at the moment. Please let me know if you find any errors.

**Comment by sovit-123** [comment]
@joshua-atolagbe 
Just for addition. These are the kind of results that you can expect with the current pipeline with the COCO pretrained Faster RCNN ResNet50 FPN V2.

![image_6](https://github.com/user-attachments/assets/35ead78b-70bd-4e0b-a2c3-b20b8c33daf5)

**Comment by sovit-123** [comment]
Just an update.
Updated the `sahi_inference.py` script for custom trained models as well.
Now you can use not just Torchvision pretrained models, but also custom trained weights with the script.

Let me know if I can close the issue.

**Comment by merbenf** [comment]
@sovit-123 Thanks a lot Sir. I'm very grateful for your support and effort. I also sent you a connection request and a message on LinkedIn asking for your opinion. I would greatly appreciate it if you could take a moment to respond.

**Comment by joshua-atolagbe** [comment]
> @joshua-atolagbe Thanks. Just for some clarification. It is the inference script. It will give you the FPS on the folder of images (the directory path that you will pass that contains the images). However, it does not carry out the metric calculations. It is quite difficult to integrate SAHI with object detection evaluation. Although it sounds plausible to add that, it is not very straightforward.
> 
> However, the image inference with SAHI is in working state at the moment. Please let me know if you find any errors.

I have some codes that can compute these metrics when using SAHI inference. I will make a pull request soon.

**Comment by joshua-atolagbe** [comment]
> Just an update. Updated the `sahi_inference.py` script for custom trained models as well. Now you can use not just Torchvision pretrained models, but also custom trained weights with the script.
> 
> Let me know if I can close the issue.

Great work man👏. Yes you can close the issue

**Comment by sovit-123** [comment]
Thanks.
